rpm/src/main/java/org/eclipse/packager/rpm/signature/RpmFileSignatureProcessor.java

```diff
@@ -0,0 +1,226 @@
+            processor.feedHeader(headerBuf.slice());
+            processor.feedPayloadData(payloadBuf.slice());
+            processor.finish(signatureHeader);
+        }
+        ByteBuffer signatureBuf = Headers.render(signatureHeader.makeEntries(), true, Rpms.IMMUTABLE_TAG_SIGNATURE);
+        final int payloadSize = signatureBuf.remaining();
+        final int padding = Rpms.padding(payloadSize);
+        byte[] signature = safeReadBuffer(signatureBuf);
+        ByteArrayOutputStream result = new ByteArrayOutputStream();
+        result.write(signature);
+        if (padding > 0) {
+            result.write(safeReadBuffer(ByteBuffer.wrap(Rpms.EMPTY_128, 0, padding)));
+        }
+        return result.toByteArray();
+    }
+
+    /**
+     * <p>
+     * Safe read (without buffer bytes) the given buffer and return it to a byte
+     * array
+     * </p>
+     * 
+     * @param buf : the {@link ByteBuffer} to read
+     * @return byte array
+     */
+    private static byte[] safeReadBuffer(ByteBuffer buf) {
+        ByteArrayOutputStream result = new ByteArrayOutputStream();
+        while (buf.hasRemaining()) {
+            result.write(buf.get());
+        }
+        return result.toByteArray();
+    }
+
+    /**
+     * <p>
+     * Return all default {@link SignatureProcessor} defined in
+     * {@link SignatureProcessors}
+     * </p>
+     * 
+     * @return {@link List<SignatureProcessor>} of {@link SignatureProcessor}
+     */
+    private static List<SignatureProcessor> getDefaultsSignatureProcessors() {
+        List<SignatureProcessor> signatureProcessors = new ArrayList<>();
+        signatureProcessors.add(SignatureProcessors.size());
+        signatureProcessors.add(SignatureProcessors.sha256Header());
+        signatureProcessors.add(SignatureProcessors.sha1Header());
+        signatureProcessors.add(SignatureProcessors.md5());
+        signatureProcessors.add(SignatureProcessors.payloadSize());
+
+        return signatureProcessors;
+    }
+
+    /**
+     * <p>
+     * Convert an array of bytes into a ByteBuffer
+     * </p>
+     * 
+     * @param data : byte array to convert
+     * @return a {@link ByteBuffer} built with data
+     * @throws IOException
+     */
+    private static ByteBuffer bufBytes(byte[] data) throws IOException {
+        ByteBuffer buf = ByteBuffer.allocate(data.length);
+        ReadableByteChannel headerChannel = Channels.newChannel(new ByteArrayInputStream(data));
+        IOUtils.readFully(headerChannel, buf);
+        return buf;
+    }
+
+    /**
+     * <p>
+     * Parse the byte[] to an RpmInputStream
+     * </p>
+     * 
+     * @param buf : byte array representing the rpm file
+     * @return {@link RpmInputStream}
+     * @throws IOException
+     */
+    private static RpmInputStream getRpmInputStream(byte[] buf) throws IOException {
+        try (RpmInputStream ref = new RpmInputStream(new ByteArrayInputStream(buf))) {
+            ref.available(); // init RpmInputStream
```

Comment:
Can someone explain why this call to `available()` is even necessary.

Now I think the method is not correct since the stream would be closed before you use it. In any case, if you have a method which returns a stream, it becomes unclear whose job it is to close it. In this case, it would have to be the caller's and not the method itself, which is why I suggested just replacing it inline.

Reply:
The method `available()` in RpmInputStream call the method `ensureInit()` wich will read the Rpm and set all properties. Normally we don’t need this because `ensureInit()` is called everytime you need to access a property of RpmInputStream. Because I'm not using the RpmInputStream to read the Rpm file but just to get somes informations, I just init all properties and I close the stream.

Reply:
Yea, I assumed that was the case :) … I think having this explanation in the comment would be helpful to let people understand.